src/OpenTelemetry.Instrumentation.StartupHook/AssemblyResolver.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace OpenTelemetry.Instrumentation.StartupHook
+{
+    internal class AssemblyResolver
+    {
+        internal static string OtelManagedProfilerPath => Environment.GetEnvironmentVariable("OTEL_DOTNET_TRACER_HOME");
+
+        public static Assembly LoadAssemblyFromSharedLocation(AssemblyLoadContext context, AssemblyName assemblyName)
+        {
+            string sharedAssemblyPath = null;
+            try
+            {
+                // TODO: If dotnet does not like the provided assembly, it will retry this event in a loop.
+                // Add a logic to avoid cyclic loop.
+                sharedAssemblyPath = Path.Combine(OtelManagedProfilerPath, $"{assemblyName.Name}.dll");
```

Comment:
This code is very similar to the `Managed.Loader` likely it can be combined. That said I think it is better to do it in a separate PR. Perhaps after adding the tests suggested by Robert.

Reply:
When we move on to DiagnosticSource Activity proxy, this part will go through a big change to address the context. We will use LoadFile instead of LoadFrom. It's better to combine this as a part of DiagnosticSource proxy work.

Reply:
@rajkumar-rangaraj this next step is interesting: how to use the OTel SDK when using the Activity proxy? I guess that it comes in the next PR :)